serverless/client/client.go

```diff
@@ -47,7 +48,9 @@ import (
 // based implementation MUST return this error when it receives a 404 StatusCode.
 type Fetcher func(ctx context.Context, path string) ([]byte, error)
 
-func fetchCheckpointAndParse(ctx context.Context, f Fetcher, v note.Verifier) (*log.Checkpoint, []byte, error) {
+// FetchCheckpoint retrieves and opens a checkpoint from the log.
+// Returns both the parsed structure and the raw serialised checkpoint.
+func FetchCheckpoint(ctx context.Context, f Fetcher, v note.Verifier) (*log.Checkpoint, []byte, error) {
```

Comment:
Have you tried this on a checkpoint that has been co-signed? `note.Open` will return an error if there exist any signatures that don't check out. It may be worth noting this in the description. It could even be a TODO. Eventually this should take the log sig verifier, and parse providing that matches. It could also return (either as another argument, or in an error) the list of unverified sigs.

Reply:
I don't think that's true...  `note.Open` will err if _none_ of the sigs are valid or we didn't pass in a matching `Verifier` (https://play.golang.org/p/U6FJqvl0ZkD)

My thinking here is that this is explicitly fetching a checkpoint from a specific log - so it MUST be signed by the log, anything else (e.g. witness sigs) is a concern for somewhere else.  There might be a better construction which we can move to, but I was trying to strike a balance between convenience and stopping folks from getting it too wrong (e.g. doing a note.Open with {logA, Witness} verifiers on a checkpoint signed by {LogE, Witness} would open without error but is probably not what you would expect here)